src/json.hpp

```diff
@@ -1623,7 +1625,7 @@ class lexer
     };
 
     /// return name of values of type token_type (only used for errors)
-    static const char* token_type_name(const token_type t) noexcept
+    static const value_t* token_type_name(const token_type t) noexcept
```

Comment:
This will fail if `const value_t *` is not compatible with the simple string literals returned by this function.

Reply:
Hi germar. I'm not insisting for these chanes to be pulled in. I'm not so skilled in templates but I will. Anyways are you agree that once the string type is declared as a class that is possible to change then all the library contents must follow this rule? If not then at least 3rd arg of basic_json makes no sence. And if so 1st, 2rd and the allocator make no sence too. So we have a bug! As for me it was enough to solve my problem. If you are able to fix this - fix it, please.

Reply:
And as for more why do you suppose and limit the users that text have to be only as char compatible types?
Do not understand me wrong I want your library to the best.

Reply:
FYI, I'm just a user of the library.  As I understand it, the internal parsing is done using plain `char`, while the storage of the data in the `json` object uses the defined string type.  @nlohmann can provide more details.  There may be a smaller set of changes needed to do what you need, not sure.  My concern is simply that while it may work for your case, there may be other users that use a different string type that uses something like `wchar_t` instead, which would cause this to break.  I can't remember if someone was able to do that or not, just pointing out potential issues.

Reply:
Yes, internally, we use `char` in the lexer class. This is independent of the string class used in the library. I am not sure what this PR tries to achieve. Could we maybe have an issue before where a bug or a desired extension is discussed?

Reply:
If this is the best way to get desired chenges - yes, sure! What should I do? I'm newbie here. If thi requires a sample I can made it but tomorrow(simply drunk ;) - having beer )

Reply:
In any case, I have yet to understand what changes you desire.